src/utils.js

```diff
@@ -0,0 +1,68 @@
+import {
+  interpolateSpectral,
+  scaleSequential,
+  scaleLinear,
+  timeDays,
+  range,
+} from 'd3';
+
+export function createColorGenerator(min_value, max_value, color) {
+  let colorGenerator;
+  let colorCode = color;
+  const hexRegex = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/;
+  const hexResult = hexRegex.test(colorCode);
```

Comment:
remove the hex regex, css accepts string like `blue`, `red`, etc too

Reply:
@maneetgoyal sir the hex code is being used by the `scale linear` to map linear relation ship between `min_value, max_value` to color code `#ffffff` to the `colorCode that user will provide`. So using color names like `red, blue, white` won't work.

Reply:
let me research about it sir, it may work.

Reply:
@maneetgoyal sir, It is working fine with correct color names, but the problem will arise when the user will provide a wrong name for a color. for ex: blus, at such situation the heatmap will always be of color black.
![Screenshot from 2022-04-29 11-12-52](https://user-images.githubusercontent.com/79472606/165890482-af9ea7ef-9fef-4057-a9a3-e12c49598b09.png)